<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1107</width>
    <height>566</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1107</width>
    <height>566</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1107</width>
    <height>566</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>NIR_Fedosov_Dmitrii_Dmitrievich_PRIb-231</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::SystemSearch"/>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Main Window Styles */
MainWindow {
    background-color: #f8f9fa;
    font-family: 'Segoe UI', 'Helvetica Neue', sans-serif;
}

#centralwidget {
    background-color: white;
    border-radius: 16px;
    margin: 16px;
    padding: 20px;
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.08);
}

/* Input Field */
#Input_file_path {
    border: 2px solid #e0e0e0;
    border-radius: 12px;
    padding: 12px 16px;
    font-size: 14px;
    color: #333;
    background: #fafafa;
    selection-background-color: #4d7cff;
    selection-color: white;
    transition: all 0.3s ease;
}

#Input_file_path:focus {
    border-color: #4d7cff;
    background: white;
    box-shadow: 0 0 0 3px rgba(77, 124, 255, 0.1);
}

/* Graphics View */
#graphicsView {
    background-color: white;
    border-radius: 16px;
    border: 2px solid #f0f0f0;
    margin: 16px 0;
    transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
}

#graphicsView:hover {
    border-color: #e0e0e0;
    box-shadow: 0 8px 24px rgba(0, 0, 0, 0.06);
}

/* Labels */
#label_from_N, #label_to_N, #label_ves_N, #label_InputFilePath {
    font-size: 14px;
    font-weight: 600;
    color: #555;
    letter-spacing: 0.3px;
}

#label_from, #label_to, #label_ves {
    font-size: 15px;
    font-weight: 700;
    color: #222;
    background: linear-gradient(to right, #f5f5f5, #fafafa);
    border-radius: 8px;
    padding: 6px 12px;
    min-width: 40px;
    text-align: center;
    border: 1px solid #eee;
}

/* Error Label */
#label_error {
    color: #ff4757;
    font-size: 13px;
    font-weight: 500;
    background: rgba(255, 71, 87, 0.05);
    border-radius: 8px;
    padding: 10px 14px;
    margin: 8px 0;
    border-left: 3px solid #ff4757;
    animation: fadeIn 0.4s ease-out;
}

/* Horizontal Line */
#line {
    background: linear-gradient(to right, transparent, #e0e0e0, transparent);
    height: 1px;
    margin: 20px 0;
    border: none;
}

/* Plus/Minus Buttons */
#pushButton_from_plus, #pushButton_to_plus,
#pushButton_from_minus, #pushButton_to_minus {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #f8f8f8, stop:1 #e8e8e8);
    border-radius: 10px;
    width: 36px;
    height: 36px;
    border: none;
    color: #4d7cff;
    font-size: 18px;
    font-weight: 800;
    transition: all 0.2s cubic-bezier(0.4, 0, 0.2, 1);
    box-shadow: 0 2px 6px rgba(0, 0, 0, 0.05);
}

#pushButton_from_plus:hover, #pushButton_to_plus:hover,
#pushButton_from_minus:hover, #pushButton_to_minus:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 #ffffff, stop:1 #f0f0f0);
    color: #3a6bf0;
    transform: translateY(-2px);
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
}

#pushButton_from_plus:pressed, #pushButton_to_plus:pressed,
#pushButton_from_minus:pressed, #pushButton_to_minus:pressed {
    transform: translateY(0);
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.08);
}

/* Draw Basic Graph Button */
#pushButton_draw_basic_graf {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, 
                stop:0 #6e8efb, stop:1 #4d7cff);
    border: none;
    border-radius: 12px;
    padding: 12px 24px;
    font-weight: 600;
    font-size: 14px;
    color: white;
    text-transform: uppercase;
    letter-spacing: 0.5px;
    box-shadow: 0 4px 16px rgba(77, 124, 255, 0.3);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

#pushButton_draw_basic_graf:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, 
                stop:0 #7d9bfb, stop:1 #5a8aff);
    transform: translateY(-2px);
    box-shadow: 0 6px 20px rgba(77, 124, 255, 0.4);
}

#pushButton_draw_basic_graf:pressed {
    transform: translateY(0);
    box-shadow: 0 2px 8px rgba(77, 124, 255, 0.3);
}

/* Generate Path Button */
#pushButton_generate_path {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, 
                stop:0 #FFD700, stop:1 #FFAA00);
    border: none;
    border-radius: 12px;
    padding: 12px 24px;
    font-weight: 700;
    font-size: 14px;
    color: #222;
    text-transform: uppercase;
    letter-spacing: 0.5px;
    box-shadow: 0 4px 18px rgba(255, 170, 0, 0.3);
    transition: all 0.3s cubic-bezier(0.4, 0, 0.2, 1);
}

#pushButton_generate_path:hover {
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, 
                stop:0 #FFE600, stop:1 #FFC000);
    transform: translateY(-2px);
    box-shadow: 0 6px 22px rgba(255, 170, 0, 0.4);
}

#pushButton_generate_path:pressed {
    transform: translateY(0);
    box-shadow: 0 2px 10px rgba(255, 170, 0, 0.3);
}

/* Layouts */
#horizontalLayout, #horizontalLayout_2, #horizontalLayout_3 {
    spacing: 12px;
    margin: 12px 0;
}

/* Animations */
@keyframes fadeIn {
    from {
        opacity: 0;
        transform: translateY(-8px);
    }
    to {
        opacity: 1;
        transform: translateY(0);
    }
}

@keyframes pulse {
    0% { transform: scale(1); }
    50% { transform: scale(1.05); }
    100% { transform: scale(1); }
}

/* Hover effect for buttons */
QPushButton:hover {
    animation: pulse 1.5s infinite;
}</string>
  </property>
  <property name="iconSize">
   <size>
    <width>24</width>
    <height>24</height>
   </size>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonStyle::ToolButtonIconOnly</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>661</width>
      <height>551</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>3</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="label_InputFilePath">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>30</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>DemiBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>Введите путь к файлу:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_draw_basic_graf">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>110</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>DemiBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>Нарисовать граф по умолчанию</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>200</y>
      <width>201</width>
      <height>43</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>3</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_from_N">
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <fontweight>DemiBold</fontweight>
        </font>
       </property>
       <property name="text">
        <string>Откуда:</string>
       </property>
       <property name="margin">
        <number>4</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_from">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton_from_plus">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>200</y>
      <width>40</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>ExtraBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_from_minus">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>200</y>
      <width>40</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>ExtraBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_to_minus">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>270</y>
      <width>40</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>ExtraBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_to_plus">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>270</y>
      <width>40</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <fontweight>ExtraBold</fontweight>
     </font>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>270</y>
      <width>201</width>
      <height>43</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>3</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_to_N">
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <fontweight>DemiBold</fontweight>
        </font>
       </property>
       <property name="text">
        <string>Куда:</string>
       </property>
       <property name="margin">
        <number>4</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_to">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton_generate_path">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>350</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>-1</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Построить маршрут</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>420</y>
      <width>281</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_ves_N">
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <fontweight>DemiBold</fontweight>
        </font>
       </property>
       <property name="text">
        <string>Вес маршрута:</string>
       </property>
       <property name="margin">
        <number>9</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_ves">
       <property name="font">
        <font>
         <pointsize>-1</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_error">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>500</y>
      <width>401</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>60</y>
      <width>381</width>
      <height>24</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(188, 188, 188);
color: rgb(0, 0, 0);</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
